docs/developer/index.asciidoc

```diff
@@ -2,6 +2,9 @@
 = Developer guide
 
 --
+
+NOTE: This is our legacy developer guide, and while we strive to keep it accurate, new content is added inside the {kib-repo}blob/{branch}/dev_docs[Kibana repo]. Internal developers can access this content rendered in our https://docs.elastic.dev/kibana-dev-docs/getting-started/welcome[Kibana Developer Guide]. External developers can access this information from the repository, however links won't work. We hope to eventually make the rendered guide public for the benefit of external developers.
```

Comment:
I attempted to simplify the statement about the rendered docs being available only internally by adding the phrase "at the moment" which alludes to our intention of making this public eventually.

```suggestion
NOTE: This is our legacy developer guide, and while we strive to keep it accurate, new content is added inside the {kib-repo}blob/{branch}/dev_docs[Kibana repo]. The rendered [guide](https://docs.elastic.dev/kibana-dev-docs/getting-started/welcome[Kibana Developer Guide] can only be accessed internally at the moment, though the raw content is public in our {kib-repo}blob/{branch}/dev_docs[public repository].
```